src/sources/kafka.rs

```diff
@@ -270,6 +290,16 @@ async fn kafka_source(
         }
     }
 
+    match consumer.commit_consumer_state(CommitMode::Sync) {
```

Comment:
I haven't looked, is there any way to check if there are any offsets to store before calling this, instead of catching the error after? It looks harmless the way you've written it, just a little surprising to have a no-op treated as an error case.

Reply:
I don't see one in the rdkafka crate docs. You could maybe get this by comparing the results of `consumer.position()` and `consumer.committed_offsets()`?

Reply:
`TopicPartitionList` implements `PartialEq` so something like this would probably work:

```rust
    match consumer.position() {
        Ok(position) => {
            if position.elements().iter().all(|el| el.offset() == Offset::Invalid) {
                info!("No offsets assigned yet (i.e. nothing has been consumed)");
            } else if let Ok(committed_position) = consumer.committed(Duration::from_secs(5)) {
                if position != committed_position {
                    warn!("Need to commit offsets: {:?} => {:?}", position, committed_position);
                    // Call commit_consumer_state here
                } else {
                    info!("Looks like everything has been committed");
                }
            } else {
                warn!("Oops, something happened while looking up committed offset positions");
            }
        },
        Err(e) => {
            warn!("Oops, error while looking up the consumer offset positions: {:?}", e);
        }
    }
```

I'm inclined to just try committing and no-op the `NoOffset` error variant, but the above could probably be packaged into a `has_offsets_to_commit` function if we did want to go that route.

Reply:
Right, this looks like more trouble than it's worth. I'd agree with your inclination.